### PR TITLE
[Arctic-329] fix mysql/ams-init.sql for optimize_task_history

### LINKED ISSUE #329: [Bug]: mysql init sql error
### What happened?

when create table optimize_task_history
```
ERROR 1171 (42000): All parts of a PRIMARY KEY must be NOT NULL; if you need NULL in a key, use UNIQUE instead
```

### Affects Versions

master/0.3.1

### What engines are you seeing the problem on?

AMS

### How to reproduce

in mysql client
```
source src/main/resources/sql/mysql/ams-init.sql
```

### Relevant log output

```shell
ERROR 1171 (42000): All parts of a PRIMARY KEY must be NOT NULL; if you need NULL in a key, use UNIQUE instead
```

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

fix #329

## Brief change log

  - move optimize_task_history upgrade sql to src/main/resources/sql/mysql/upgrade-0.3.0-to-0.3.1.sql

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate
image
image

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by hameizi** [comment]
I think we only need to update ams-init.sql because this exception was introduced in v0.3.1. And ams-init.sql is not executable.

File: ams/ams-server/src/main/resources/sql/mysql/ams-init.sql [hunk 1]
```diff
@@ -237,7 +237,6 @@ CREATE TABLE `optimize_task_history`
     `cost_time`       bigint(20) DEFAULT NULL COMMENT 'Task cost time',
     `queue_id`        int(11) DEFAULT NULL COMMENT 'Task queue id',
     `task_group_id`   varchar(40) DEFAULT NULL COMMENT 'Task group id',
-    PRIMARY KEY (`task_history_id`,`task_group_id`),
```
Comment (zhoujinsong) [inline ams/ams-server/src/main/resources/sql/mysql/ams-init.sql:240]: Why remove this primary key?

Comment (wangtaohz) [reply ams/ams-server/src/main/resources/sql/mysql/ams-init.sql:240]: NOT NULL column is required for primary key, but `task_history_id` and `task_group_id` are DEFAULT NULL.

**Comment by wangtaohz** [comment]
>

**Comment by hameizi** [comment]
> > I think we only need to update ams-init.sql because this exception was introduced in v0.3.1. And ams-init.sql is not executable.
> 
> Shall we just change `task_history_id` and `task_group_id` from DEFAULT NULL to NOT NULL in ams-init.sql directly?
> 
> I would have thought adding into upgrade-0.3.0-to-0.3.1.sql was more appropriate.

it's ok

**Review by zhoujinsong — APPROVED** [review]

**Pull request merged**